AEPAnalytics/Sources/AnalyticsState.swift

```diff
@@ -268,19 +268,24 @@ class AnalyticsState {
         batchLimit = AnalyticsConstants.Default.BATCH_LIMIT
         launchHitDelay = AnalyticsConstants.Default.LAUNCH_HIT_DELAY
         backDateSessionInfoEnabled = AnalyticsConstants.Default.BACKDATE_SESSION_INFO_ENABLED
-        marketingCloudOrganizationId = nil
         analyticForwardingEnabled = AnalyticsConstants.Default.FORWARDING_ENABLED
+        lifecycleMaxSessionLength = AnalyticsConstants.Default.LIFECYCLE_MAX_SESSION_LENGTH
+        lifecycleSessionStartTimestamp = AnalyticsConstants.Default.LIFECYCLE_SESSION_START_TIMESTAMP
+        assuranceSessionActive = AnalyticsConstants.Default.ASSURANCE_SESSION_ENABLED
+        rsids = nil
+        host = nil
+        resetIdentities()
+    }
+
+    /// Clears all identities.
+    func resetIdentities() {
+        marketingCloudOrganizationId = nil
```

Comment:
We don't have to reset OrgId.

Reply:
But should we treat it pretty much as the same as opted out?

Reply:
I think this mid will be reset by Identity and we will get the shared state update with empty value which will reset it in AA as well, right @praveek?

Reply:
The resetIdentities event should reset the data related to current identity, such as ecid, locationHint, blob, visitorIdsList, advertisingId, vid, aid. Additionally the queued hits can be discarded.
It is not necessary to reset the configuration fields and lifecycle/app information.

When handling the next Analytics event, the Identity shared state should be fetched to retrieve the latest identity and populate these fields again.